src/main/java/data/Vendor.java

```diff
@@ -93,15 +111,17 @@ public void addAccount(Account account) {
   /** Return a JSON String representing a billing config's content. */
   public String buildJsonConfig() {
     ArrayList<Account> accounts = getAccounts();
-    String config = String.format("{\"legacy_vendor_id\":%s," +
-      "\"next_gen_vendor_id\":%d,\"accounts\":[", getLegacyVendorID(), 
+    String config = String.format("{\"legacy_customer_id\":\"%s\"," +
+      "\"next_gen_customer_id\":%d,\"accounts\":[", getLegacyVendorID(),
       getNextGenVendorID());
 
-    for(int i = 0; i < accounts.size(); i++) {
-      config += accounts.get(i).toJson();
+    if (!accounts.isEmpty()) {
+      for (int i = 0; i < accounts.size() - 1; i++) {
+        config += accounts.get(i).buildJsonConfig() + ",";
+      }
+      config += accounts.get(accounts.size() - 1).buildJsonConfig();
```

Comment:
This would be a nice place to use some nifty Java tricks.

Could simply be: config += String.join(",", accounts.stream().map(Account::buildJsonConfig).collect(Collectors.toList()));

That would make this a lot cleaner